dashboard/client/src/pages/job/JobDetail.tsx

```diff
@@ -60,7 +60,8 @@ export const JobDetailChartsPage = () => {
   const { data } = useSWR(
     "useDataDatasets",
     async () => {
-      const rsp = await getDataDatasets();
+      // Only display details for Ray Datasets that belong to this job.
+      const rsp = await getDataDatasets(job?.job_id ?? null);
```

Comment:
please make this change:

```
const { data } = useSWR(
    job?.job_id ? ["useDataDatasets", job.job_id] : null,
    async ([_, jobId]) => {
      // Only display details for Ray Datasets that belong to this job.
      const rsp = await getDataDatasets(jobId);
```

This makes 2 changes:
1. It doesn't send a request to fetch datasets unless we have a job id. The alternate behavior of fetching without a filter will produce a weird unexpected output.
2. It adds the jobId to the SWR key. This makes it so SWR will use a different cache key if the job id changes. This will make it so if you switch between job pages, the data from diferrent jobs won't be loaded temporarily before fetching the new API.